src/txmempool.cpp

```diff
@@ -451,8 +451,13 @@ void CTxMemPoolEntry::UpdateAncestorState(int64_t modifySize, CAmount modifyFee,
     assert(int(nSigOpCostWithAncestors) >= 0);
 }
 
-CTxMemPool::CTxMemPool(CBlockPolicyEstimator* estimator, int check_ratio)
-    : m_check_ratio(check_ratio), minerPolicyEstimator(estimator)
+CTxMemPool::CTxMemPool(const Options& opts)
+    : m_check_ratio(opts.check_ratio),
+      minerPolicyEstimator(opts.estimator),
+      m_max_size(opts.max_size_bytes),
+      m_expiry(std::chrono::hours{opts.expiry}),
+      m_full_rbf(opts.full_rbf),
```

Comment:
This design is straightly inherited from #25290 where it has been given a chance to be discussed and reviewed extensively. If you think we would be better to store a copy of `Options` in `CTxMemPool`, I'll invite you to let thoughts there and explain why storing a copy of `Options` is better. It can be addressed as a follow-up.